docs/source/notes/gradcheck.rst

```diff
@@ -0,0 +1,223 @@
+.. _gradcheck-mechanics:
+
+Gradcheck mechanics
+===================
+
+This note presents an overview of how the :meth:`~torch.autograd.gradcheck` and :meth:`~torch.autograd.gradgradcheck` functions work.
+
+It will cover both forward and backward mode AD for both real and complex-valued functions as well as higher-order derivatives.
+This note also covers both the default behavior of gradcheck as well as the case where :code:`fast_mode=True` argument is passed (referred to as fast gradcheck below).
+
+.. contents:: :local:
+    :depth: 2
+
+Notations and background information
+------------------------------------
+
+Throughout this note, we will use the following convention:
+1. :math:`x`, :math:`y`, :math:`a`, :math:`b`, :math:`v`, :math:`u`, :math:`ur` and :math:`ui` as real-valued vectors and :math:`z` is a complex-valued vector that can be rewritten in terms of two real-valued vectors as :math:`z = a + i b`.
```

Comment:
I think you might need spaces between the 1, 2, etc. or else the formatting gets messed up